pt/endpoints/powerdata/resource.py

```diff
@@ -141,164 +161,78 @@ def data_table(self, limit, offset, time, field):
-                [data.uuid for data in PowerData.query.filter(
-                    PowerData.updated_at.between(start_time, end_time),
-                    PowerData.field == field,
-                    PowerData.data_type == 'Demand'
-                ).all()]
-            )
-        ).all()
-        # PV sum
-        pv_data = db.session.query((func.sum(PV.pac) / 60).label('sum')).filter(
-            PV.uuid.in_(
-                [data.uuid for data in PowerData.query.filter(
-                    PowerData.updated_at.between(start_time, end_time),
-                    PowerData.field == field,
-                    PowerData.data_type == 'PV'
-                ).all()]
-            )
-        ).all()
-        # EV sum
-        ev_data = db.session.query((func.sum(EV.power_display) / 60).label('sum')).filter(
-            EV.uuid.in_(
-                [data.uuid for data in PowerData.query.filter(
-                    PowerData.updated_at.between(start_time, end_time),
-                    PowerData.field == field,
-                    PowerData.data_type == 'EV'
-                ).all()]
-            )
-        ).all()
-        # ESS sum
-        ess_data = db.session.query((func.sum(ESS.power_display) / 60).label('sum')).filter(
-            ESS.uuid.in_(
-                [data.uuid for data in PowerData.query.filter(
-                    PowerData.updated_at.between(start_time, end_time),
-                    PowerData.field == field,
-                    PowerData.data_type == 'ESS'
-                ).all()]
-            )
-        ).all()
-        # WT sum
-        wt_data = db.session.query((func.sum(WT.windgridpower) / 60).label('sum')).filter(
-            WT.uuid.in_(
-                [data.uuid for data in PowerData.query.filter(
-                    PowerData.updated_at.between(start_time, end_time),
-                    PowerData.field == field,
-                    PowerData.data_type == 'WT'
-                ).all()]
-            )
-        ).all()
-        # record sum to data object
-        data = {
-            "Demand": round(demand_data[0].sum, 3),
-            "WT": round(wt_data[0].sum, 3),
-            "PV": round(pv_data[0].sum, 3),
-            "EV": round(ev_data[0].sum, 3),
-            "ESS": round(ess_data[0].sum, 3)
-        }
-        # Add Power generation field
+        # referencing the powerdata_datatype object
+        powerdata_datatype = self.powerdata_datatype
+        # gather date summary by traversing the power_datatype objects
+        data = {}
+        for data_type in powerdata_datatype:
+            # the kW record per minute should divide by 60 to convert to kWh
+            data[data_type] = round(db.session.query(
+                (func.sum(powerdata_datatype[data_type]['field']) / 60).label('sum')
+            ).filter(
+                powerdata_datatype[data_type]['model'].updated_at.between(start_time, end_time),
+                PowerData.field == field
+            ).all()[0][0], 3)
```

Comment:
add a comment to explain why `[0][0]`
If there is only one column, can use `first()` at here.